### PR TITLE
API-2400: Go through the handover checklist for 'api-example-scala-client"

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: app/connectors/ApiConnector.scala [hunk 1]
```diff
@@ -35,18 +37,24 @@ trait ApiConnector {
...
-      case None => Seq("Accept" -> versionHeader)
+    val authorizationHeaders: Seq[(String, String)] = token match {
+      case Some(t) => Seq("Authorization" -> s"Bearer $t")
+      case None => Seq()
     }
-    extractJson[JsValue](
-      WS.url(s"$serviceUrl$endpoint").withHeaders(headers:_*).get()
+    val headers = authorizationHeaders ++ Seq(
+      "Accept" -> versionHeader,
+      "User-Agent" -> ApplicationContext.appName,
```
Comment (gokyo) [inline app/connectors/ApiConnector.scala]: Remove these two headers from here
This is just an example client, so it does not have to abide the HMRC standards

File: app/connectors/OAuth20Connector.scala [hunk 2]
```diff
@@ -16,14 +16,15 @@
 
 package connectors
 
+import java.util.UUID
```
Comment (gokyo) [inline app/connectors/OAuth20Connector.scala]: not needed

File: app/connectors/ApiConnector.scala [hunk 3]
```diff
@@ -35,18 +37,24 @@ trait ApiConnector {
...
+      "X-Request-ID" -> s"govuk-tax-${UUID.randomUUID().toString}"
```
Comment (gokyo) [inline app/connectors/ApiConnector.scala]: remove

File: app/connectors/ApiConnector.scala [hunk 4]
```diff
@@ -16,6 +16,8 @@
 
 package connectors
 
+import java.util.UUID
```
Comment (gokyo) [inline app/connectors/ApiConnector.scala]: not needed

File: app/connectors/OAuth20Connector.scala [hunk 5]
```diff
@@ -55,14 +56,19 @@ trait OAuth20Connector {
...
-    val requestToken = WS.url(tokenUrl).withHeaders("Csrf-Token" -> "nocheck").
-      post(
-        Map(
-          "client_id" -> Seq(clientId),
-          "client_secret" -> Seq(clientSecret)
-        ) ++ body)
+    val request = WS.url(tokenUrl).withHeaders(
+      "Csrf-Token" -> "nocheck",
+      "User-Agent" -> ApplicationContext.appName,
+      "X-Request-ID" -> s"govuk-tax-${UUID.randomUUID().toString}"
```
Comment (gokyo) [inline app/connectors/OAuth20Connector.scala]: same as above

File: app/connectors/OAuth20Connector.scala [hunk 6]
```diff
@@ -55,14 +54,17 @@ trait OAuth20Connector {
...
   private def oauth2(body: Map[String, Seq[String]]): Future[OauthResponse] = {
-    val requestToken = WS.url(tokenUrl).withHeaders("Csrf-Token" -> "nocheck").
-      post(
-        Map(
-          "client_id" -> Seq(clientId),
-          "client_secret" -> Seq(clientSecret)
-        ) ++ body)
+    val request = WS.url(tokenUrl).withHeaders(
+      "Csrf-Token" -> "nocheck"
+    )
```
Comment (tom59) [inline app/connectors/OAuth20Connector.scala]: need to remove the csrf-token header. it should work without it

Comment (gokyo) [inline app/connectors/OAuth20Connector.scala]: done

**Pull request merged**